### PR TITLE
Allow prefixing simple tokens with `#`

### LINKED ISSUE #6716: Simple Tokens not allowing # before simple token
### Affected version(s)

4.13.x

### Description

Putting a # in front of an simple token will hinder replacing the simple token.

Example: ###order_id## will not be replaced (explected output for example #5300)

Tested in:
Contao 4.13.35
PHP 8.2.13

This was also discussed in Slack: https://contao.slack.com/archives/CK4J0KNDB/p1705315711120819

### PR DESCRIPTION
**Pull request by Toflar**
Fixes https://github.com/contao/contao/issues/6716

### CODE REVIEW

**Comment by Toflar** [comment]
Hmm, seems like `#` was allowed in legacy token names so we probably cannot fix this in 4.13.
We could fix it in 5, though. Wdyt @ausi ?

**Comment by leofeyer** [comment]
> Hmm, seems like `#` was allowed in legacy token names

What do you mean?

**Comment by Toflar** [comment]
See failing legacy tests

> There was 1 failure:

1) Contao\CoreBundle\Tests\String\SimpleTokenParserTest::testParsesSimpleTokensLegacy with data set "Test token replacement with special characters (#)" ('This is my ##e#mail##', array('test@foobar.com'), 'This is my test@foobar.com')
Failed asserting that two strings are identical.
--- Expected
+++ Actual
@@ @@
-'This is my test@foobar.com'
+'This is my ##e#mail##'



So seems like `#` in the token was actually allowed. So either we come up with a different regex or this can only change in 5.

**Comment by leofeyer** [comment]
I see.

File: core-bundle/src/String/SimpleTokenParser.php [hunk 1]
```diff
@@ -118,7 +118,7 @@ private function replaceTokens(string $subject, array $data): string
     {
         // Replace tokens
         return preg_replace_callback(
-            '/##([^=!<>\s]+?)##/',
+            '/##([^#=!<>\s]+?)##/',
```
Comment (ausi) [inline core-bundle/src/String/SimpleTokenParser.php]:
```suggestion
            '/##([^#=!<>\s][^=!<>\s]*?)##/',
```
Only disallow the `#` at the beginning?

**Comment by Toflar** [comment]
@ausi found the most obvious fix 🤣

**Review by fritzmg — APPROVED** [review]

**Comment by leofeyer** [comment]
Thank you @Toflar.

**Pull request merged**